From 6c2f51d9a51b17e075cb4c41a03deb625d4f42cc Mon Sep 17 00:00:00 2001
From: Fuzzing on a Shoestring <fuzzing.on.a.shoestring@gmail.com>
Date: Tue, 13 Sep 2016 13:14:01 +0000
Subject: [PATCH] Don't randomize dictionary keys during fuzzing.

---
 dict.c | 3 ++-
 hash.c | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/dict.c b/dict.c
index 5f71d55..d8e1a24 100644
--- a/dict.c
+++ b/dict.c
@@ -38,7 +38,8 @@
  *  list we will use the BigKey algo as soon as the hash size grows
  *  over MIN_DICT_SIZE so this actually works
  */
-#if defined(HAVE_RAND) && defined(HAVE_SRAND) && defined(HAVE_TIME)
+#if defined(HAVE_RAND) && defined(HAVE_SRAND) && defined(HAVE_TIME) &&  \
+    !defined(FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION)
 #define DICT_RANDOMIZATION
 #endif
 
diff --git a/hash.c b/hash.c
index 0145109..da9c758 100644
--- a/hash.c
+++ b/hash.c
@@ -33,7 +33,8 @@
  * it seems that having hash randomization might be a good idea
  * when using XML with untrusted data
  */
-#if defined(HAVE_RAND) && defined(HAVE_SRAND) && defined(HAVE_TIME)
+#if defined(HAVE_RAND) && defined(HAVE_SRAND) && defined(HAVE_TIME) &&  \
+    !defined(FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION)
 #define HASH_RANDOMIZATION
 #endif
 
-- 
2.7.4

